gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilExtension.kt

```diff
@@ -16,15 +21,15 @@ open class AnvilExtension {
    *
    * By default this feature is disabled.
    */
-  var generateDaggerFactories = false
+  val generateDaggerFactories: Property<Boolean> = objects.property(Boolean::class.java)
```

Comment:
nit: I think this could be `abstract val generateDaggerFactories: Property<Boolean>` and Gradle will take care of instantiating that for you. I usually only bother to do that myself when I use a pattern like:
```kotlin
private val prop = objects.property(Boolean::class.java).convention(false)
```
i.e., when I'm setting a default value.

(It doesn't really matter, though. Gradle doing it for you and you using the `ObjectFactory` are totally equivalent.)

Reply:
Good point re: convention, added in #288

the reason I couldn't use the abstract val is actually interesting. Gradle itself will complain with an esoteric message about that overload with a diff return type when trying to create the class because it doesn't seem to take this into account that the JVM allows for overloaded return types. Declaring it manually takes that out of Gradle's hands